Update index.md

dalejrodriguez opened the pull request:
Adding section to authenticate to APK's using long lived tokens. Based upon prospect escalation here:

https://github.com/chainguard-dev/customer-issues/issues/2357

[ ] Check if this is a typo or other quick fix and ignore the rest :)

## Type of change
documentation

### What should this PR do?
adds instructions on how to authenticate to an apk registry using a pull token

### Why are we making this change?
Addresses prospect issue here - https://github.com/chainguard-dev/customer-issues/issues/2357

### What are the acceptance criteria? 
Just make sure you can follow the steps and it works

### How should this PR be tested?
After complete the steps are completed, run a simple command like the following to verify your token works:

(Use an image with a shell/apk)

docker run --rm -it  -e HTTP_AUTH --entrypoint /bin/sh cgr.dev/$ORGINIZATION/$IMAGE:$TAG-dev 

apk update

Here is an example of my test:

jovan@Dales-MacBook-Pro ~ % docker run --rm -it  -e HTTP_AUTH --entrypoint /bin/sh cgr.dev/dale.rodriguez/chainguard-base
/ # apk update
fetch https://apk.cgr.dev/de767c7cdef05a49e0433dc083cd9473c01bc836/aarch64/APKINDEX.tar.gz
WARNING: updating and opening https://apk.cgr.dev/de767c7cdef05a49e0433dc083cd9473c01bc836: Permission denied
fetch https://virtualapk.cgr.dev/de767c7cdef05a49e0433dc083cd9473c01bc836/sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855/chainguard/aarch64/APKINDEX.tar.gz
fetch https://virtualapk.cgr.dev/de767c7cdef05a49e0433dc083cd9473c01bc836/sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855/extra-packages/aarch64/APKINDEX.tar.gz
 [https://virtualapk.cgr.dev/de767c7cdef05a49e0433dc083cd9473c01bc836/sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855/chainguard]
 [https://virtualapk.cgr.dev/de767c7cdef05a49e0433dc083cd9473c01bc836/sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855/extra-packages]
2 unavailable, 0 stale; 118974 distinct packages available

dalejrodriguez committed:
Update index.md

Adding section to authenticate to APK's using long lived tokens. Based upon prospect escalation here:

https://github.com/chainguard-dev/customer-issues/issues/2357

Signed-off-by: Dale J. Rodriguez <37818051+dalejrodriguez@users.noreply.github.com>

mosabua commented on content/chainguard/chainguard-images/features/private-apk-repos/index.md:
```diff
@@ -107,6 +107,36 @@ apk update
+
+In this example we will setup a pull token that expires in 35 days, to be used with CI. 
+
+**Generate a Pull Token:**
```
```suggestion
**Generate a pull token:**
```

mosabua reviewed (commented):
Some minor suggestions

dalejrodriguez committed:
Update content/chainguard/chainguard-images/features/private-apk-repos/index.md

Co-authored-by: Manfred Moser <manfred@simpligility.ca>
Signed-off-by: Dale J. Rodriguez <37818051+dalejrodriguez@users.noreply.github.com>

dalejrodriguez committed:
Update content/chainguard/chainguard-images/features/private-apk-repos/index.md

Co-authored-by: Manfred Moser <manfred@simpligility.ca>
Signed-off-by: Dale J. Rodriguez <37818051+dalejrodriguez@users.noreply.github.com>

dalejrodriguez committed:
Update content/chainguard/chainguard-images/features/private-apk-repos/index.md

Co-authored-by: Manfred Moser <manfred@simpligility.ca>
Signed-off-by: Dale J. Rodriguez <37818051+dalejrodriguez@users.noreply.github.com>

mosabua commented on content/chainguard/chainguard-images/features/private-apk-repos/index.md:
````diff
@@ -107,6 +107,36 @@ apk update
+**Generate a pull token:**
+
+```shell
+chainctl auth pull-token --parent=<YOUR_CHAINGUARD_ORG> --ttl=850h -o json > /tmp/token.json
````
Not sure about this but I think this should support the env output option and then this would be better - see https://edu.chainguard.dev/chainguard/libraries/access/#use-environment-variables  .. can you please check and if that works update the docs

amouat replied:
Both here and later replace <YOUR_CHAINGUARD_ORG> with ORGANIZATION for consistency with the rest of the document.

mosabua reviewed (changes requested):
Needs testing of --output env and potentially update

amouat commented on content/chainguard/chainguard-images/features/private-apk-repos/index.md:
```diff
@@ -107,6 +107,36 @@ apk update
 
+## Generating a Pull Token for Authentication 
+
+To generate a longer lived token for authentication purposes, it is necessary to create a pull token to access your private APK repository. 
```
maybe "custom pull token" or "pull token with a custom TTL value"?

amouat reviewed (approved):
This looks good!

Many thanks for the PR. Just a minor comment about consistency.

dalejrodriguez commented:
@mosabua It looks like the the instructions for the build tools worked for this example, let me know if you can verify this and I will update the PR with these instructions. They are much more cleaner. 

jovan@Dales-MacBook-Pro ~ % chainctl auth pull-token --library-ecosystem=java --parent=dale.rodriguez
To use this pull token in another environment, supply the following for Basic authorization:

Username: `de767....`

Password: `eSy0...`

jovan@Dales-MacBook-Pro ~ % chainctl auth pull-token --output env --library-ecosystem=java --parent=dale.rodriguez
export CHAINGUARD_JAVA_IDENTITY_ID=`de767....`
CHAINGUARD_JAVA_TOKEN=`eyJhb...`

jovan@Dales-MacBook-Pro ~ % export HTTP_AUTH=basic:apk.cgr.dev:user:${CHAINGUARD_JAVA_IDENTITY_ID}:${CHAINGUARD_JAVA_TOKEN}

Testing:

jovan@Dales-MacBook-Pro ~ % !3089
docker run --rm -it  -e HTTP_AUTH --entrypoint /bin/sh cgr.dev/dale.rodriguez/chainguard-base

`/ # apk update
fetch https://apk.cgr.dev/de767c7cdef05a49e0433dc083cd9473c01bc836/aarch64/APKINDEX.tar.gz
WARNING: updating and opening https://apk.cgr.dev/de767c7cdef05a49e0433dc083cd9473c01bc836: Permission denied
fetch https://virtualapk.cgr.dev/de767c7cdef05a49e0433dc083cd9473c01bc836/sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855/chainguard/aarch64/APKINDEX.tar.gz
fetch https://virtualapk.cgr.dev/de767c7cdef05a49e0433dc083cd9473c01bc836/sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855/extra-packages/aarch64/APKINDEX.tar.gz
 [https://virtualapk.cgr.dev/de767c7cdef05a49e0433dc083cd9473c01bc836/sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855/chainguard]
 [https://virtualapk.cgr.dev/de767c7cdef05a49e0433dc083cd9473c01bc836/sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855/extra-packages]
2 unavailable, 0 stale; 119720 distinct packages available`

dalejrodriguez commented:
@amouat Good call outs I will modify the PR with that information.

dalejrodriguez committed:
Update index.md

Added some additional information and changed doc based upon Adrian and Manfred's comments. 

Signed-off-by: Dale J. Rodriguez <37818051+dalejrodriguez@users.noreply.github.com>

dalejrodriguez reviewed (commented):
Modifications made.

mosabua commented:
@dalejrodriguez my question would be if it works without the ecosystem part. I tested this now. It works:

```
chainctl auth pull-token --output env
Opening browser to https://issuer.enforce.dev/oauth?audience=https%3A%2F%2Fconsole-api.enforce.dev&client_id=auth0&connection=google-oauth2&create_refresh_token=true&exit=redirect&redirect=http%3A%2F%2Flocalhost%3A59032%2Fcallback%3Ftoken%3Dtrue&skip_registration=true

   ✔ Selected folder chainguard.edu.
export CHAINGUARD_IDENTITY_ID=45a0c61ea6fd977f050c5fb9ac06a69eed764595/8a42a4c194e6341f
export CHAINGUARD_TOKEN=eyJhbGciOiJSUzI1NiJ9.e...
```

note the env var names

mosabua commented:
Replaced by https://github.com/chainguard-dev/edu/pull/2402
